libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -196,6 +201,77 @@ public static synchronized void reset(UserAccount account) {
         }
     }
 
+    /**
+     * Stop the sync manager
+     * It might take a while for active syncs to actually get stopped
+     * Call isStopped() to see if syncManager is fully paused
+     */
+    public void stop() {
+        stopRequested = true;
+        for (SyncTask syncTask : activeSyncs.values()) {
+            syncTask.stop();
+        }
+    }
+
+    /**
+     * @return true if stop was requested but there are still active syncs
+     */
+    public boolean isStopping() {
+        return stopRequested && activeSyncs.size() > 0;
+    }
+
+    /**
+     * @return true if stop was requested and there no syncs are active anymore
+     */
+    public boolean isStopped() {
+        return stopRequested && activeSyncs.size() == 0;
```

Comment:
Once we stop it, wouldn't we reset the value of `stopRequested`, i.e. `stopRequested = false`? Should this condition be just `return activeSyncs.size() == 0`, i.e. no active syncs running?

Reply:
The idea was once the SyncManager is stopped, it won't accept new jobs (and will try to wrap up existing jobs as quickly as possible). The app has to call resume to allow jobs to be submitted again. If I flip the flag back, jobs submission won't be prevented.

Reply:
Gotcha.